examples/repo_example/succinct_hash_bin_delegations.py

```diff
@@ -0,0 +1,201 @@
+succinct_roles.keyids.append(key.keyid)
+delegations_keys_info[key.keyid] = key
+
+# We are ready to define the Delegations instance which we will add to targets.
+# As mentioned, standard roles are not allowed together with succinct_roles.
+delegations = Delegations(
+    delegations_keys_info, roles=None, succinct_roles=succinct_roles
+)
+
+targets.signed.delegations = delegations
+
+# Delegated targets (bins) roles
+# ------------------------------
+# We have defined the top-level targets metadata instance which utilizes
+# succinct_roles. With succinct_roles we have defined the bins number, common
+# bin properties and keys information, but we haven't actually created the
+# bins targets metadata instances.
+
+# mypy linter requires that we verify that succinct_roles is not None
+assert targets.signed.delegations.succinct_roles is not None
+
+# We can get all bin names for a SuccinctRoles instance with get_roles()
+delegated_bins: Dict[str, Metadata[Targets]] = {}
+for delegated_bin_name in targets.signed.delegations.succinct_roles.get_roles():
+    delegated_bins[delegated_bin_name] = Metadata(
+        Targets(expires=expiration_date)
+    )
+
+# Add target file inside a delegated role (bin)
+# ---------------------------------------------
+# For the purpose of this example we will protect the integrity of this very
+# example script by adding its file info to the corresponding bin metadata.
+
+# NOTE: See "Targets" paragraph in 'basic_repo.py' example for more details
+# about adding target file infos to targets metadata.
+local_path = Path(__file__).resolve()
+target_path = f"{local_path.parts[-2]}/{local_path.parts[-1]}"
+target_file_info = TargetFile.from_file(target_path, str(local_path))
+
+# We don't know yet in which delegated role (bin) our target belongs.
+# With SuccinctRoles.get_role_for_target() we can get the name of the delegated
+# role (bin) responsible for that target_path.
+target_bin = targets.signed.delegations.succinct_roles.get_role_for_target(
+    target_path
+)
+
+# In our example with NUMBER_OF_BINS = 32 and the current file as target_path
+# the target_bin is "delegated_bin-0d"
+
+# Now we can add the current target to the bin responsible for it.
+delegated_bins[target_bin].signed.targets[target_path] = target_file_info
+
+# Sign and persist
+# ----------------
+# Sign all metadata and persist to a temporary directory at CWD for review using
+# versioned file names. Most notably see 'targets.json' and
+# 'delegated_bin-0d.json'. For more information on versioned file names see:
+# https://theupdateframework.github.io/specification/latest/#writing-consistent-snapshots
+
+# NOTE: See "Persist metadata" paragraph in 'basic_repo.py' example for more
+# details about serialization formats and metadata file name convention.
+PRETTY = JSONSerializer(compact=False)
+TMP_DIR = tempfile.mkdtemp(dir=os.getcwd())
+
+# Generate a key for targets we haven't added one up to this point.
```

Comment:
Does this phrase a conjunction? Should it be two sentences?

Reply:
Agreed. It's a weird phrase.
```suggestion
# Generate a key for targets that  up to this point have not received one.
```

Reply:
I suggest to move the `create_key` call up to where we created the other key. Then we don't have to explain why we all of a sudden create another key.